mono/metadata/object.c

```diff
@@ -4395,76 +4395,43 @@ mono_runtime_try_run_main (MonoMethod *method, int argc, char* argv[],
 	return mono_runtime_try_exec_main (method, args, exc);
 }
 
-
 static MonoObject*
-serialize_object (MonoObject *obj, gboolean *failure, MonoObject **exc)
+serialize_or_deserialize_object (MonoObject *obj, MonoError *error,
+	const gchar *method_name, MonoMethod **method)
 {
-	static MonoMethod *serialize_method;
-
-	ERROR_DECL (error);
-	void *params [1];
-	MonoObject *array;
-
-	if (!serialize_method) {
+	if (!*method) {
 		MonoClass *klass = mono_class_get_remoting_services_class ();
-		serialize_method = mono_class_get_method_from_name (klass, "SerializeCallData", -1);
+		*method = mono_class_get_method_from_name (klass, method_name, -1);
 	}
 
-	if (!serialize_method) {
-		*failure = TRUE;
+	if (!*method) {
+		mono_error_set_exception_instance (error, NULL);
 		return NULL;
```

Comment:
What will this do with a NULL argument ? I think this could be just an assert.

Reply:
I'm emulating the previous behavior I believe -- where it'd set *failure = TRUE and leave exc == NULL and caller would mono_error_set_exception_instance.

Reply:
I think "it works" -- the error handling infrastructure will take it -- but maybe indeed assert is preferred.